sdk/core/Azure.Core/src/TokenRequestContext.cs

```diff
@@ -96,5 +122,28 @@ public TokenRequestContext(string[] scopes, string? parentRequestId = default, s
         /// If you don't handle CAE responses in these API calls, your app could end up in a loop retrying an API call with a token that is still in the returned lifespan of the token but has been revoked due to CAE.
         /// </remarks>
         public bool IsCaeEnabled { get; }
+
+         /// <summary>
+        /// Indicates whether to enable Proof of Possession (PoP) for the requested token.
+        /// </summary>
+        public bool IsProofOfPossessionEnabled { get; }
+
+        /// <summary>
+        /// The nonce value required for PoP token requests. This is typically retrieved from teh WWW-Authenticate header of a 401 challenge response.
+        /// This is used in combination with <see cref="ResourceRequestUri"/> and <see cref="ResourceRequestMethod"/> to generate the PoP token.
+        /// </summary>
+        public string? ProofOfPossessionNonce { get; }
+
+        private readonly Request? _request;
+
+        /// <summary>
+        /// The HTTP method of the request. This is used in combination with <see cref="ResourceRequestUri"/> and <see cref="ProofOfPossessionNonce"/> to generate the PoP token.
+        /// </summary>
+        public HttpMethod? ResourceRequestMethod => new(_request!.Method.ToString());
+
+        /// <summary>
+        /// The URI of the request. This is used in combination with <see cref="ResourceRequestMethod"/> and <see cref="ProofOfPossessionNonce"/> to generate the PoP token.
+        /// </summary>
+        public Uri? ResourceRequestUri => _request?.Uri.ToUri();
```

Comment:
Is the prefix `Resource` needed here?

Reply:
Initially I didn't have this prefix, but I added it thinking that it is more descriptive of which Uri is expected here. For example it could be the token endpoint given the context of the type.

Reply:
Also, should these new properties be considered when token caching here, https://github.com/Azure/azure-sdk-for-net/blob/44a61f96c1b0a69770666ce8e2144fe2d7f201d6/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L386?

Reply:
That policy is specific to Bearer tokens currently, so it should be agnostic.